java/arcs/android/devtools/storage/DevToolsStorageConnection.kt

```diff
@@ -0,0 +1,57 @@
+package arcs.android.devtools.storage
+
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import arcs.android.storage.ParcelableStoreOptions
+import arcs.sdk.android.storage.service.StorageService
+import arcs.sdk.android.storage.service.StorageServiceBindingDelegate
+import arcs.sdk.android.storage.service.StorageServiceConnection
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+typealias DevToolsConnectionFactory = () -> StorageServiceConnection
+
+/**
+ * Returns a [DevToolsConnectionFactory] implementation which uses the provided [context] to bind to
+ * the [StorageService] and the provided [coroutineContext] as the parent for
+ * [StorageServiceConnection.connectAsync]'s [Deferred] return value.
+ */
+@Suppress("FunctionName")
+@ExperimentalCoroutinesApi
+fun DevToolsConnectionFactory(
+    context: Context,
+    storageClass: Class<StorageService>,
+    bindingDelegate: StorageServiceBindingDelegate = DevToolsStorageManagerBindingDelegate(
+        context,
+        storageClass
+    ),
+    coroutineContext: CoroutineContext = Dispatchers.Default
+): DevToolsConnectionFactory = { StorageServiceConnection(bindingDelegate, null, coroutineContext) }
```

Comment:
Nice!

If this works out well then I think it's great.  One more way it could be done is to look up the StorageService using the Android APIs which let you investigate the app's manifest - but even that approach would likely still use this Factory (it'd just mean that the developer wouldn't have to configure as much)

Reply:
Updated to the getStorageClass() method, see details in comment below.